.travis.yml

```diff
@@ -23,90 +17,124 @@ notifications:
     on_success: always
     on_failure: always
 
-before_script:
-  - export NODE_OPTIONS=--max-old-space-size=4096
-
-script:
-  - set -e
-  - npm_config_mode=yes npx lockfile-lint --type npm --path package.json --validate-https --allowed-hosts npm
-  - npm run lint-ci
-  - npm run build
-  - travis_retry npm run test-ci
-  - npm run test-e2e-ci
-
-before_deploy:
-  # Workaround to run before_deploy only once
-  - >
-    if ! [ "$TAG" ]; then
-      pip install --user awscli
-      # Put AWS in path
-      export PATH=$PATH:$HOME/.local/bin
-      # Login to AWS ECR, credentials defined in $AWS_ACCESS_KEY_ID and $AWS_SECRET_ACCESS_KEY
-      $(aws ecr get-login --no-include-email --region ap-southeast-1)
-      export TAG=travis-$TRAVIS_COMMIT-$TRAVIS_BUILD_NUMBER
-      docker build -f Dockerfile.production -t $REPO:$TAG .
-      docker tag $REPO:$TAG $REPO:$TRAVIS_BRANCH
-      docker push $REPO
-      # Add TAG to Dockerrun
-      sed -i -e "s/@TAG/$TAG/g" Dockerrun.aws.json 
-      zip -r "$TAG.zip" .ebextensions Dockerrun.aws.json
-    fi
-  - export ELASTIC_BEANSTALK_LABEL="$TAG-$(env TZ=Asia/Singapore date "+%Y%m%d%H%M%S")"
+jobs:
+  include:
+    - stage: Build application
+      install: npm ci
+      cache:
+        - npm
+        - pip
+      before_script:
+        - export NODE_OPTIONS=--max-old-space-size=4096
+      script:
+        - set -e
+        - npm_config_mode=yes npx lockfile-lint --type npm --path package.json --validate-https --allowed-hosts npm
+        - npm run lint-ci
+        - npm run build
+      workspaces:
+        create:
+          name: build
+          paths: .
+    - stage: Tests
+      name: Javascript tests
+      workspaces:
+        use: build
+      script:
+        - travis_retry npm run test-backend-jasmine
+        - npm run test-frontend
+    - name: Typescript tests
+      workspaces:
+        use: build
+      script:
+        - travis_retry npm run test-backend-jest
+      after_success:
+        - cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js
+    - name: End-to-end tests
+      workspaces:
+        use: build
+      addons:
+        chrome: stable
+      script:
+        - npm run test-e2e-ci
```

Comment:
these 3 jobs run in parallel, right?

Reply:
Yes, but it ends up being (slightly) slower (right now) since Travis is limited to 2 concurrent jobs. Probably what happened when we last looked at it. But with 5 jobs all can finally run at once and then should cut down a couple minutes